apps/color-picker/src/locations/Field.tsx

```diff
@@ -0,0 +1,148 @@
+    left: 0,
+  }),
+};
+
+// Dropdown + margin top + box shadow
+const HEIGHT_DEFAULT = 41;
+const HEIGHT_BASE = 52 + 4 + 14;
+const HEIGHT_ITEM = 36;
+
+type FieldValue = Color | string;
+
+const Field = () => {
+  const sdk = useSDK<FieldExtensionSDK>();
+  const [isOpen, setIsOpen] = useState(false);
+  const [value, setValue] = useFieldValue<FieldValue>();
+  const customColorPicker = useRef<HTMLInputElement>(null);
+
+  const storeHexValue = sdk.field.type === 'Symbol';
+  const allowCustomValue = sdk.parameters.instance.withCustomValue;
+
+  // @ts-ignore
+  const theme: Theme = sdk.parameters.installation.themes[0];
+
+  useEffect(() => {
+    if (!isOpen) {
+      sdk.window.updateHeight(HEIGHT_DEFAULT);
+      return;
+    }
+
+    // @ts-ignore
+    const customItemHeight = sdk.parameters.instance.withCustomValue
+      ? HEIGHT_ITEM
+      : 0;
+    const calculatedHeight =
+      HEIGHT_BASE + customItemHeight + theme.colors.length * HEIGHT_ITEM;
+
+    sdk.window.updateHeight(calculatedHeight <= 400 ? calculatedHeight : 400);
+  }, [isOpen, sdk, theme]);
+
+  const name = useMemo(() => {
+    if (storeHexValue) {
+      const color = theme.colors.find((c) => c.value === (value as string));
+      if (color) {
+        return color.name;
+      }
+      if (allowCustomValue) {
+        return 'Custom';
+      } else {
+        return 'Invalid';
+      }
+    } else {
+      return (value as Color).name ?? 'Invalid';
+    }
+  }, [allowCustomValue, storeHexValue, theme.colors, value]);
+
+  return (
+    <Form>
+      <Menu
+        isOpen={isOpen}
+        onOpen={() => setIsOpen(true)}
+        onClose={() => setIsOpen(false)}
+      >
+        <Menu.Trigger>
+          <Button endIcon={<ChevronDownIcon />} isFullWidth>
+            <Flex alignItems="center" gap="spacingXs">
+              <span
+                className={styles.colorBox(
+                  (typeof value === 'string' ? value : value?.value) ??
+                    '#ffffff'
+                )}
+              />
+              <Flex gap="spacing2Xs">
+                {name}{' '}
+                <span className={styles.hexValue}>
+                  {(typeof value === 'string' ? value : value?.value) ?? ''}
+                </span>
+              </Flex>
+            </Flex>
+          </Button>
+        </Menu.Trigger>
+        <Menu.List className={styles.menuList}>
+          {theme.colors.map((color: Color) => (
+            <Menu.Item
+              key={color.id}
+              onClick={() => setValue(storeHexValue ? color.value : color)}
+            >
+              <Flex alignItems="center" gap="spacingXs">
+                <span className={styles.colorBox(color.value)} />
+                {color.name}
+              </Flex>
+            </Menu.Item>
+          ))}
+          {allowCustomValue && (
+            <Menu.Item onClick={() => customColorPicker?.current?.click()}>
+              Other...
```

Comment:
Should this be "Custom" as we are selecting a custom value

Reply:
Yes, makes sense!